### DISCUSSION TITLE
Azure Quick Review difference to Azure Review Checklists

### DISCUSSION THREAD

**Discussion by mschwrdtnr** [Azure/azqr#148] [Q&A] [answered]
Hi people,

thumbs up for the project! I am searching for an easy way to review Azure Environments against Azure Best Practices. Beside this project I found the comprehensive review checklists of the FTA team (https://github.com/Azure/review-checklists/tree/main). 

I've found that the review checklists are more comprehensive and includes much more checks for different purposes. They also have a way to query Azure environments and include the results in a spreadsheet. But like your project the queries don't check for everything. I am wondering what are the differences between this project and the other one? Are you working together or by your own? 

Thanks!

**Comment by mschwrdtnr** [comment]
Found another project which also checks for rules based on the WAF: https://azure.github.io/PSRule.Rules.Azure/en/rules/

**Reply by mschwrdtnr** [reply]
This project is very interesting because it has the same purpose as azqr. The setup is quite the same but `PSRule.Azure` are checking for more rules. The advantage of `azqr` is the easy to use vsc output. `PSRule.Azure` also have an csv output with more checks but not so easy to read and filter for resource groups isn't possible.

**Comment by cmendible** [comment, marked as answer]
`azqr` was created to address a very specific need I had back in 2022. Initially, I had to run three assessments to get a clear picture of various solutions in terms of SLAs, use of Availability Zones, and Diagnostic Settings. At the time, I was not aware of the existence of the `review-checklist` or `PSRule.Azure`.

When some of my peers saw the assessments I was able to deliver with the early bits of `azqr`, they asked me to add more checks (rules) and change the output format from markdown to Excel.

Also, many of my customers work in very restrictive environments. Therefore, the ability to run a self-contained binary, in any OS, that requires just read-only permissions was very important.

Moving forward to 2023, based on great feedback from both peers and customers, we moved the original repo to the `Azure` organization, added support for more services, fixed some issues and even added a Power BI template.

To be open we are giving `azqr` a "best effort" approach, and since we learned about `PSRule.Azure`, we are slowly trying to catch up with their great set of rules, with no intention to replace @BernieWhite's and collaborators outstanding work.

IMHO, given the fact that each solution is built using different technologies and have a different set of features, there is room for both.

**Reply by mschwrdtnr** [reply]
@cmendible Thanks for your clarification! The big question for me is how the best of both (`azqr` and `PSRule.Azure`) can be brought together. 

**For me, the following points are the great features of the respective solution:** _I see both tools from the view of an in-flight environment review._

**AZQR**
- Overview Page of all Services with SKU, SLA, AZ, PVT, DS and CA
- Defender Overview Page
- Azure Advisor Recommendations
- Costs Overview Page
- Native filter for subscription, resource group and location

**PSRule.Azure**
- More rules (and that's a very big advantage)


For me it looks like that both tools uses other ways to evaluate the rules. `PSRule.Azure` exports the current environment as json while `azqr` runs graph requests on the fly. The exported data from `PSRule.Azure` should also allow presenting all the information as the `azq` report. Except Advisor, Defender and Costs. 

For me personally and the use case I want to fulfill (Azure Review on the fly) AZQR dashboards with the checks of PSRule.Azure would nice. @cmendible do you also see this use case and the need to find a way using the rules in AZQR? If so, please let me know you thoughts about and a way to accomplish. I would be happy to participate! 🥳

**Reply by cmendible** [reply]
@mschwrdtnr We are working on adding more rules and of course we are open to contributions 😊
